NAS-112510 / 21.10 / Filter svclb from pod selector

Ornias1993 opened the pull request:
This PR removes all "Service Loadbalancer" pods from the log and shell selector.
These pods are created by the K3S ServiceLoadbalancer and their logs have no value for the user.

They also lead to significant confusion, with users submitting the wrong logs.

**Before:**
---
![image](https://user-images.githubusercontent.com/7613738/134503451-6eceec18-e611-469e-9d12-71fd67296e7c.png)

![image](https://user-images.githubusercontent.com/7613738/134503471-22546436-c333-497a-bc8c-f7aa3db10e2a.png)



**After:**
---
![image](https://user-images.githubusercontent.com/7613738/134503288-6c6c0c20-fdd2-4cf9-b221-1e1492ff13a6.png)

![image](https://user-images.githubusercontent.com/7613738/134503488-71183708-1cf6-4f17-9ef0-755bd0b0070e.png)

bugclerk commented:
Jira URL: https://jira.ixsystems.com/browse/NAS-112510

undsoft commented on src/app/pages/applications/chart-releases/chart-releases.component.ts:
```diff
@@ -501,6 +501,15 @@ export class ChartReleasesComponent implements OnInit {
 
+  filterPodNames(podList: string[]): string[] {
+    for (const entry of podList) {
+      if (entry.includes('svclb')) {
```
Let's add a comment explaining what this does.

undsoft commented on src/app/pages/applications/chart-releases/chart-releases.component.ts:
```diff
@@ -501,6 +501,15 @@ export class ChartReleasesComponent implements OnInit {
   }
 
+  filterPodNames(podList: string[]): string[] {
+    for (const entry of podList) {
```
You can probably just do
```
return podList.filter((pod) => pod.includes('svclb'));
```

undsoft reviewed (changes requested):
It looks good, but let's also wait for Waqar's approval.

Ornias1993 commented:
> It looks good, but let's also wait for Waqar's approval.

Thanks for your feedback!
I've removed the function and directly filtered the podnames like you showed. Also added a comment to make clear why it's done.

I've passed this past Waqar the 06-sept  and got no objection (while discussing https://github.com/truenas/middleware/pull/7481 ) :)

sonicaj commented:
@Ornias1993 this is not required as middleware is not showing the pods spawned from daemonsets.
